### PR TITLE
Adding `HumanitZ`

### PR DESCRIPTION
**Pull request by vp-en**
# Description

Adding the steam game `HumanitZ` as a Pterodactyl Egg. Solves https://github.com/parkervcp/eggs/issues/2624

The included "[README.txt](https://paste.ec/paste/pSSv5A8O#Qw3FgfYY9xd5pP9h7zaHMdi7pGxlfXm8o1ux1T6Ue5P)" file says that you can use `-steamservername`, but that doesn't work at all.

I manually did the configuration file parsing, which seem to work fine. The `ServerName` variable doesn't like `""` around it, and spaces works regardless. So you can enter whatever with spacing, without using quotation marks, and [it should work fine](https://i.imgur.com/Fef7fTy.png)

## Current Egg link
If you want to test the most up to date version of this PR's egg, use this link:
https://github.com/vp-en/eggs/blob/feat-HumanitZ/game_eggs/steamcmd_servers/humanitz/egg-humanit-z.json

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
* [X] You verify that the start command applied does not use a shell script
* [X] The egg was exported from the panel

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [X] You verify that the start command applied does not use a shell script
6. [X] The egg was exported from the panel

### CODE REVIEW

**Comment by gOOvER** [comment]
-steamservername is only for the Serverlist and and its not the Servername.

**Comment by QuintenQVD0** [comment]
also not to be difficult but ENV variables are normally all uppercase

**Comment by gOOvER** [comment]
works for me in my egg without problems

steamwebhelper_WxJM1koRuO

**Comment by ImKringle** [comment]
Yeah the server name works fine 
```
"startup": ".\/TSSGame\/Binaries\/Linux\/TSSGameServer-Linux-Shipping TSSGame -log -port={{SERVER_PORT}} -queryport={{QUERY_PORT}} -steamservername={{STEAM_NAME}}"
```

**Comment by vp-en** [comment]
I think I resolved most issues:
- Fixing the ENV's to become uppercase, with underscores separating words
- Editing the Configuration parsing to accommodate new variable names
- Adding back the `-steamservername`, since it applies to the Steam server browser, and might be needed. I put it in quotes, as that was [recommend earlier](https://github.com/parkervcp/eggs/issues/2624#issuecomment-1868312166).

Let me know if there's anything else 👍

File: game_eggs/steamcmd_servers/humanitz/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+HumanitZ is a co-op, isometric, open world survival game in a world ended by the zombie outbreak. As one of the few human survivors, try to last as long as “humanly” possible. The past can’t be changed, but you can make a difference today for the future of humanity. 
+
+## Install notes
+
+The current version of the server software is slightly buggy, and as of this writing has no proper documentation.
+
+They include a `README.txt` file with the server software, that includes a description of most of the variables, but not more than that. This is only obtainable via the SteamCMD tool.
+
+The `README.txt` also specifies that you can use variables such as `-steamservername` appended to the launch command, but that doesn't work properly. Therefore this egg is designed to manually parse/edit the `GameServerSettings.ini` file, with the values entered in the variables in Pterodactyl.
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/humanitz/README.md:15]: This now can be removed?

**Review by QuintenQVD0 — APPROVED** [review]
LGTM

File: game_eggs/steamcmd_servers/humanitz/README.md (hunk 1 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/humanitz/README.md:15]: @vp-en Just this one and then it is fine for me.

Comment (vp-en) [reply game_eggs/steamcmd_servers/humanitz/README.md:15]: I just left my home. I'll be home later to fix it 😁

**Comment by ValhallaCommunityOfficial** [comment]
hmm, for some reason it's not showing the full server name. Seems like if there's a space in the server name, only the first word shows on server list. Everything else is working fine.

**Comment by gOOvER** [comment]
> hmm, for some reason it's not showing the full server name. Seems like if there's a space in the server name, only the first word shows on server list. Everything else is working fine.

SERVERNAME or SteamServerName?

**Comment by gOOvER** [comment]
> hmm, for some reason it's not showing the full server name. Seems like if there's a space in the server name, only the first word shows on server list. Everything else is working fine.

works fine for me

msedge_s7DS8E8kRb
TSSGame-Win64-Shipping_rtRhtJxo9Q

**Comment by ImKringle** [comment]
Screenshot 2023-12-25 074910
Seems to work fine for me as well, I'd make sure its set up right startup wise

**Comment by ValhallaCommunityOfficial** [comment]
Ah, I found the issue. it doesnt accept | as a spacer. The test egg did however. No worries! Thank you all for creating this egg. Been following this game for 2 years, very excited for this moment.

**Comment by ValhallaCommunityOfficial** [comment]
Server not showing up on the in-game server list after todays update on a fresh install. Was working fine previously.

**Comment by ImKringle** [comment]
> Server not showing up on the in-game server list after todays update on a fresh install. Was working fine previously. edit: shows on steam server list just not in-game. No way to join.

Linux server just got an update 5 hours ago. Try to pull that first. If it keeps up you'd need to open an [Issue ](https://github.com/parkervcp/eggs/issues) or go to the Ptero discord [here](https://discord.com/invite/pterodactyl)

**Pull request merged**